### PR TITLE
[efficiency] remove redundant calls to update all frames. only update…

### PR DESCRIPTION
**Pull request by proyan**
… the ones that are required

This reduces the time used by calc (100 nodes) by about 100us.

### CODE REVIEW

File: src/multibody/actions/free-fwddyn.cpp [hunk 1]
```diff
@@ -69,7 +69,6 @@ void DifferentialActionModelFreeFwdDynamics::calc(const boost::shared_ptr<Differ
   }
 
   // Computing the cost value and residuals
-  pinocchio::updateFramePlacements(pinocchio_, d->pinocchio);
```
Comment (cmastalli) [inline src/multibody/actions/free-fwddyn.cpp]: Could we replicate this on `contact-fwddyn.cpp`?

**Comment by proyan** [comment]
The failed builds are because of:
```
The following packages have unmet dependencies:

 robotpkg-py27-example-robot-data : Depends: robotpkg-example-robot-data (= 3.1.0) but 3.1.2 is to be installed or

                                             robotpkg-example-robot-data (= 3.0.0) but 3.1.2 is to be installed or

                                             robotpkg-example-robot-data (= 3.1.1) but 3.1.2 is to be installed

E: Unable to correct problems, you have held broken packages
```

@cmastalli Could you please have a look at the CI issues?

File: src/multibody/actions/free-fwddyn.cpp (hunk 1 above)
Comment (proyan) [reply src/multibody/actions/free-fwddyn.cpp]: I'm on the calcdiff right now :) , afterwards I'll look at contactDAM.

Comment (cmastalli) [reply src/multibody/actions/free-fwddyn.cpp]: Great! thanks!
Do not forget to share the results.

**Comment by proyan** [comment]
I checked the example logs. They are unchanged.

**Comment by proyan** [comment]
@cmastalli I think this is ready. Could you please merge this?

**Comment by cmastalli** [comment]
@proyan - It misses the changes in the contact-fwddyn. Otherwise, there is duplication of updating the frame placement.
I have pushed those changes, and after passing the CI, I will merge them.

Thank you for this nice contribution.

**Comment by proyan** [comment]
I didn't want to put the contact fwddynamics in this PR, because I didn't want to put the time to be sure it doesn't break anything. This PR, I'm sure that it doesn't.

But if we are adding the contact-fwddynamics as well, we have to consider the costs dependent on this dynamics (contact-force and contact-friction-cone), and also propagate this change to impulse dynamics, impulses, and impulse based costs.

Don't merge this yet in that case. I'll make the changes and push.

**Comment by proyan** [comment]
Did you check that it doesn't break with respect to the contact-force and contact-force friction cone costs?

**Comment by cmastalli** [comment]
Yes, I checked it.

**Comment by proyan** [comment]
Cool. Thanks.

**Pull request merged**